src/viam/sdk/config/resource.hpp

```diff
@@ -26,6 +26,11 @@ class ResourceConfig {
     viam::app::v1::ComponentConfig to_proto() const;
     ResourceConfig(std::string type);
     Name resource_name();
+    void set_api(API api);
+    void set_frame(LinkConfig frame);
+    void set_model(Model model);
+    void set_name(std::string name);
+    void set_namespace(std::string ns);
```

Comment:
Hmm what's the point of adding these setters here and on `LinkConfig` (the getters on `LinkConfig` seem fine)? I see you're using them in tests, but it mostly seems like you're calling `set_*()` and then `*()` and asserting that the field was set as expected (which hypothetically does not test any existing functionality). I would be against providing these methods publicly to users if possible.

Reply:
The setters are just for testing, so you're right when you say that these methods are probably not the way to go. I guess what I can do is make a proto object and then call `from_proto()` on it. I will change that!

Reply:
Makes sense. I was using `GeometryConfig` as a template, not realizing that there must be a reason why we don't have these setter functions elsewhere.